### PR TITLE
[qfix] Add WithEndpointChange heal option

### PR DESCRIPTION
**Pull request by Bolodya1997**
## Description
Adds `WithEndpointChange` heal option to be enabled only on NSC.

## How Has This Been Tested?

- [x] Added unit testing to cover
- [x] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [x] Bug fix
- [ ] New functionallity
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

File: pkg/networkservice/common/heal/option.go [hunk 1]
```diff
@@ -67,8 +67,16 @@ func WithRestoreTimeout(restoreTimeout time.Duration) Option {
 	}
 }
 
+// WithEndpointChange sets if Connection.EndpointName can be changed with monitor updates
+func WithEndpointChange() Option {
```
Comment (denis-tingaikin) [inline pkg/networkservice/common/heal/option.go:71]: @Bolodya1997 Could you please explain in what cases we plan do not use this option?

Comment (Bolodya1997) [reply pkg/networkservice/common/heal/option.go:71]: It is NSC-only option. Do you mean adding such comment in code?

Comment (denis-tingaikin) [reply pkg/networkservice/common/heal/option.go:71]: I don't quite follow why do we need the option only for nscs.

**Comment by denis-tingaikin** [comment]
I don't like adding more complexity into heal, but we are planning to remove and replace exist heal logic.


## Motviation

1. https://github.com/networkservicemesh/sdk/pull/1082#discussion_r713012114
2. This PR is required for https://github.com/networkservicemesh/deployments-k8s/issues/2381

So merging this as a part of solution for https://github.com/networkservicemesh/deployments-k8s/issues/2381

**Pull request merged**